### PR TITLE
fix timeout issues in SilentWebUI

### PR DESCRIPTION
**Pull request by stefangossner**
SilentWebUI always failed for me as the auth takes longer than 2 seconds.

### CODE REVIEW

**Comment by jmprieur** [comment]
@stefangossner 
thanks for your contribution.
Out of curiosity, and if this is not indiscrete, I'd like to understand where this delay come from: Which cloud are you targeting? (Black Forest? or the Azure public cloud? or ...) and where from?

**Comment by stefangossner** [comment]
Azure public Cloud I guess. From Germany. But people using my tool reported the issue also from other locations world-wide.

File: src/Microsoft.Identity.Client/Microsoft.Identity.Client.csproj [hunk 1]
```diff
@@ -23,6 +23,9 @@
   <PropertyGroup Condition="'$(TargetFramework)' == '$(TargetFrameworkNetStandard)'">
     <DefineConstants>$(DefineConstants);NETSTANDARD_RUNTIME</DefineConstants>
   </PropertyGroup>
+  <PropertyGroup>
+    <SignAssembly>true</SignAssembly>
```
Comment (bgavrilMS) [inline src/Microsoft.Identity.Client/Microsoft.Identity.Client.csproj]: This is not needed, we add it to all projects via Directory.Build.Props

File: src/Microsoft.Identity.Client/Platforms/net45/SilentWebUI.cs [hunk 2]
```diff
@@ -44,7 +44,7 @@ internal class SilentWebUI : WebUI, IDisposable
         /// This is how long all redirect navigations are allowed to run for before a graceful
         /// termination of the entire browser based authentication process is attempted.
         /// </summary>
-        private const int NavigationOverallTimeout = 2000;
+        private const int NavigationOverallTimeout = 10000;
```
Comment (bgavrilMS) [inline src/Microsoft.Identity.Client/Platforms/net45/SilentWebUI.cs]: Agreed with this one, 2s seems like a very strict timeout given there is a network call in the middle.

Comment (henrik-me) [reply src/Microsoft.Identity.Client/Platforms/net45/SilentWebUI.cs]: Makes sense to set the timeout to 10s

File: src/Microsoft.Identity.Client/Microsoft.Identity.Client.csproj (hunk 1 above)
Comment (henrik-me) [reply src/Microsoft.Identity.Client/Microsoft.Identity.Client.csproj]: pls. remove

---
In reply to: [262455207](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/939#discussion_r262455207) [](ancestors = 262455207)

**Comment by henrik-me** [comment]
@stefangossner : I would like this change to happen in dev3x as that is our next release. Will you be able to take a dependency on MSAL v3 instead of v2?

**Comment by stefangossner** [comment]
@henrik-me v3 is fine for me.

**Comment by henrik-me** [comment]
@stefangossner, Sounds good. Would be great if you can help by rebasing to dev3x and bring the PR to dev3x as that will ensure we can quickly get this integrated.

**Comment by stefangossner** [comment]
@henrik-me: here is the pull for dev3x: https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/952

**Comment by henrik-me** [comment]
Change approved and committed. closing this PR. Thanks a lot for the contribution!

**Pull request closed without merging**